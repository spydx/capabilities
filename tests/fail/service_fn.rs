use capabilities_derive::service;

struct User {
    pub name: String,
}

struct UserError;

#[service]
fn create_user() -> Result<User, UserError> {
    Ok(User { name: "Kenneth".to_string() })
}

fn main() {

}