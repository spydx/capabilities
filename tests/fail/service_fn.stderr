error: Unsupported literal, has to be a type from Capabilities lib
 --> tests/fail/service_fn.rs:9:1
  |
9 | #[service]
  | ^^^^^^^^^^
  |
  = note: this error originates in the attribute macro `service` (in Nightly builds, run with -Z macro-backtrace for more info)

error: custom attribute panicked
 --> tests/fail/service_fn.rs:9:1
  |
9 | #[service]
  | ^^^^^^^^^^
  |
  = help: message: called `Option::unwrap()` on a `None` value
